src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipValidator.java

```diff
@@ -29,36 +30,53 @@ public List<ValidationStatusError> validate(LimitedPartnershipDto limitedPartner
         violations.forEach(v ->
                 errorsList.add(createValidationStatusError(v.getMessage(), v.getPropertyPath().toString())));
 
-        // TODO These checks are valid for Partnership Types PFLP and SPFLP (the LP7D CHIPS transaction). Code will
-        //      need changing when other Partnership Types need to be validated and sent to CHIPS
-
         final var dataDto = limitedPartnershipDto.getData();
+
+        checkCommonFields(dataDto, errorsList);
+        checkTypeSpecificFields(dataDto, errorsList);
+
+        return errorsList;
+    }
+
+    private void checkCommonFields(DataDto dataDto, List<ValidationStatusError> errorsList) {
+        if (dataDto.getEmail() == null) {
+            errorsList.add(createValidationStatusError("Email is required", "data.email"));
+        }
+
+        if (dataDto.getJurisdiction() == null) {
+            errorsList.add(createValidationStatusError("Jurisdiction is required", "data.jurisdiction"));
+        }
+
+        if (dataDto.getRegisteredOfficeAddress() == null) {
+            errorsList.add(createValidationStatusError("Registered office address is required",
+                    "data.registeredOfficeAddress"));
+        }
+
+        if (dataDto.getPrincipalPlaceOfBusinessAddress() == null) {
+            errorsList.add(createValidationStatusError("Principal place of business address is required",
+                    "data.principalPlaceOfBusinessAddress"));
+        }
+    }
+
+    private void checkTypeSpecificFields(DataDto dataDto, List<ValidationStatusError> errorsList) {
         if (PartnershipType.PFLP.equals(dataDto.getPartnershipType())
                 || PartnershipType.SPFLP.equals(dataDto.getPartnershipType())) {
-            if (dataDto.getEmail() == null) {
-                errorsList.add(createValidationStatusError("Email is required", "data.email"));
-            }
-
-            if (dataDto.getJurisdiction() == null) {
-                errorsList.add(createValidationStatusError("Jurisdiction is required", "data.jurisdiction"));
+            if (dataDto.getTerm() != null) {
+                errorsList.add(createValidationStatusError("Term is not required", "data.term"));
             }
 
-            if (dataDto.getRegisteredOfficeAddress() == null) {
-                errorsList.add(createValidationStatusError("Registered office address is required",
-                        "data.registeredOfficeAddress"));
+            if (dataDto.getSicCodes() != null) {
```

Comment:
This check is new and relates to the LP7D. Probably missing as SIC codes were implemented after this end-point was. Tests added to cover this and I'll add a tester note on the story.